app/internal/page/app.go

```diff
@@ -489,11 +490,28 @@ const (
 	CanGoBack
 )
 
-func makeHandle(mux *http.ServeMux, logger Logger, store sessions.Store, localizer localize.Localizer, lang Lang, rumConfig RumConfig, staticHash string, paths AppPaths, defaultOptions handleOpt) func(string, handleOpt, Handler) {
+func makeHandle(mux *http.ServeMux, logger Logger, store sessions.Store, localizer localize.Localizer, lang Lang, rumConfig RumConfig, staticHash string, paths AppPaths, defaultOptions handleOpt, randomString func(int) string) func(string, handleOpt, Handler) {
 	return func(path string, opt handleOpt, h Handler) {
 		opt = opt | defaultOptions
 
 		mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+			csrfSession, err := store.Get(r, "csrf")
+
+			if r.Method == http.MethodPost {
+				if err != nil {
+					http.Error(w, err.Error(), http.StatusInternalServerError)
+					return
+				}
+
+				if !csrfValid(r, csrfSession) {
+					http.Error(w, "CSRF token not valid", http.StatusForbidden)
+					return
+				}
+			}
+
+			csrfSession.Values = map[interface{}]interface{}{"token": randomString(12)}
+			_ = store.Save(r, w, csrfSession)
```

Comment:
also we'll want to set `HttpOnly`, `Secure`, `SameSiteStrict`(?) and a sensible expiry (a day?) on the cookie